bundles/AdminBundle/Controller/Admin/MiscController.php

```diff
@@ -137,8 +137,18 @@ public function scriptProxyAction(Request $request)
             $allowedFileTypes = ['js', 'css'];
             $scripts = explode(',', $request->get('scripts'));
 
-            if ($request->get('scriptPath')) {
-                $scriptPath = PIMCORE_WEB_ROOT . $request->get('scriptPath');
+            $scriptPath = $request->get('scriptPath');
+            if ($scriptPath) {
+                $scriptPath = realpath(PIMCORE_WEB_ROOT . $scriptPath);
+                if(!$scriptPath) {
+                    throw $this->createNotFoundException('Directory not found!');
+                }
+
+                if(!str_starts_with($scriptPath, rtrim(str_replace('../', '', PIMCORE_PROJECT_ROOT), './'))) {
+                    throw $this->createAccessDeniedException('Scripts not found! Please do not navigate out of the web root directory!');
```

Comment:
PIMCORE_PROJECT_ROOT is not the "web root directory".
Should be changed to "project root directory"

Reply:
Is this even right? Do we really want to allow any file from the project to be included in the script?

Reply:
No, this is not right, this needs to be changed to `PIMCORE_WEB_ROOT` ... 
@robertSt7 can you please create a follow up?

Reply:
We could even further restrict it to files ending with `.js` `.css` ...

Reply:
is solved here: https://github.com/pimcore/pimcore/pull/14970